services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -78,29 +82,33 @@ const getBlocks = async params => {
 		.map(block => ({ ...block.header, payload: block.payload }));
 	if (response.meta) blocks.meta = response.meta; // TODO: Build meta manually
 
-	blocks.data = blocks.data.map(block => {
-		// TODO: Update the below params after accounts index is implemented
-		block.generatorAddress = null;
-		block.generatorUsername = null;
+	blocks.data = await BluebirdPromise.map(
+		blocks.data,
+		async block => {
+			const [{ address, username }] = await getIndexedAccountByPublicKey((block.generatorPublicKey)
+				.toString('hex'));
+			block.generatorAddress = address;
+			block.generatorUsername = username;
```

Comment:
Please consider adding null checks.
```suggestion
			block.generatorPublicKey = block.generatorPublicKey.toString('hex');
			const [response] = await getIndexedAccountByPublicKey(block.generatorPublicKey);
			block.generatorAddress = response && response.address ? response.address : undefined;
			block.generatorUsername = response && response.username ? response.username : undefined;
```